source/custom/rules/kconfig_rule.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules
+
+import (
+	"sigs.k8s.io/node-feature-discovery/source/internal/kernelutils"
+)
+
+// KconfigRule implements Rule
+type KconfigRule []string
+
+var kConfigs map[string]struct{}
+
+var searchPaths []string
+
+func (kconfigs *KconfigRule) Match() (bool, error) {
+	for _, f := range *kconfigs {
+		if _, ok := kConfigs[f]; !ok {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func init() {
+	kConfigs = make(map[string]struct{})
+	kVer, err := kernelutils.GetKernelVersion()
+	if err != nil {
+		searchPaths = []string{""}
+	} else {
+		// from k8s.io/systemr-validator used by kubeadm preflight checks
+		searchPaths = []string{
+			"/usr/src/linux-" + kVer + "/.config",
+			"/usr/src/linux/.config",
+			"/usr/lib/modules/" + kVer + "/config",
+			"/usr/lib/ostree-boot/config-" + kVer,
+			"/usr/lib/kernel/config-" + kVer,
+			"/usr/src/linux-headers-" + kVer + "/.config",
+			"/lib/modules/" + kVer + "/build/.config",
```

Comment:
I think we should use the same logic in kernel feature source and here. Prolly move these to kernelutils(?)

Reply:
moved to kernelutils